### PR TITLE
Feature: Theming

### PR DESCRIPTION
**Pull request by zsh28**
made theming using context and diffrentiated dark and light styles in seperate files, added background colours for dark and light mode

### CODE REVIEW

File: App.tsx [hunk 1]
```diff
@@ -1,20 +1,64 @@
...
+        <Stack.Navigator
+          initialRouteName={"Home"}
+          screenOptions={{
+            headerRight: () => (
+              <Ionicons
+                name={theme === "light" ? "sunny" : "moon"}
+                onPress={toggleTheme}
+                size={20}
+                style={{ marginRight: 15 }}
+                color={theme === "light" ? "black" : "white"}
```
Comment (SwyftPain) [inline App.tsx]: For consistency, this could go into styles too, so its all in one place and easier to find

File: Screens/Settings.tsx [hunk 2]
```diff
@@ -0,0 +1,29 @@
+import { View, Text, TouchableOpacity } from "react-native";
+import { useContext } from "react";
+import { useNavigation } from "@react-navigation/native";
+import MadeButton from "../components/button";
+import Header from "../components/header";
+import ThemeContext from "../theme/ThemeContext";
+import { LightStyles, DarkStyles } from "../Styles";
+
+
+//needs style on #18
```
Comment (SwyftPain) [inline Screens/Settings.tsx]: ?? What does this mean? Is this comment even needed?

File: Styles/DarkStyles.tsx [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+const styles = StyleSheet.create({
+  container: {
+    flex: 1,
+    backgroundColor: "#090b38",
+    alignItems: "center",
+    justifyContent: "center",
+  },
+
+  //header
```
Comment (SwyftPain) [inline Styles/DarkStyles.tsx]: No need to comment the obvious, i am aware some of it is from previous PR but can be changed

File: Types/typeconfig.tsx [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+  Settings: undefined;
+};
+
+export type HomePropsType = NativeStackScreenProps<StackScreens, "Home">;
+export type SettingsPropsType = NativeStackScreenProps<
+  StackScreens,
+  "Settings"
+>;
+
+//Header Types
```
Comment (SwyftPain) [inline Types/typeconfig.tsx]: Everything above this is not needed, neither is footerProp since there is no footer.
I missed it in the first PR

**Review by SwyftPain — CHANGES_REQUESTED** [review]
There are some inconsistencies from the previous PR that i missed that could be updated. But there is color styling on navigation that is not consistent with the styles file and is instead in App.tsx. That may make it confusing to find and change.

**Comment by SwyftPain** [comment]
Conflicts are fine, the changes are just too big, i can deal with it myself

File: Screens/Settings.tsx (hunk 2 above)
Comment (zsh28) [reply Screens/Settings.tsx]: this is from the previous pr

File: Styles/DarkStyles.tsx [hunk 5]
```diff
@@ -40,6 +38,9 @@ const styles = StyleSheet.create({
     fontSize: 24,
     fontWeight: "bold",
   },
+  iconcolor: {
```
Comment (SwyftPain) [inline Styles/DarkStyles.tsx:41]: camelCase inconsistency but can be fixed in the next PR. Will fix conflicts myself later.

**Review by SwyftPain — APPROVED** [review]
Only small camelCase inconsistency but is fine, can be dealt with in next theming PR. Will fix the conflicts myself bit later. Good to go otherwise.

**Review by avrey-s — APPROVED** [review]
I don't see any problems or errors!

**Pull request merged**